### DISCUSSION TITLE
[Lesson5] brownie downloaded successfully but cannot use "brownie" command

### DISCUSSION THREAD

**Discussion by Code1688** [smartcontractkit/full-blockchain-solidity-course-py#1147] [Q&A]
I installed the brownie by using the below command
```
python3 -m pip install —user pipx
python3 -m pipx ensurepath

pipx install eth-brownie
```

It already be installed.
```
xxxx-MBP:brownie_simple_storagexxx$ pipx install eth-brownie
'eth-brownie' already seems to be installed. Not modifying existing installation in
'/Users/xxx/.local/pipx/venvs/eth-brownie'. Pass '--force' to force installation.
```
Successfully installed but i when i tried to brownie commend its not working as expected. The following error occurs
```
xxx-MBP:brownie_simple_storage xxx$ brownie
Traceback (most recent call last):
  File "/Users/xxx/.local/bin/brownie", line 5, in <module>
    from brownie._cli.__main__ import main
  File "/Users/xxx/.local/pipx/venvs/eth-brownie/lib/python3.10/site-packages/brownie/__init__.py", line 6, in <module>
    from brownie.project import compile_source, run
  File "/Users/xxx/.local/pipx/venvs/eth-brownie/lib/python3.10/site-packages/brownie/project/__init__.py", line 3, in <module>
    from .main import (  # NOQA 401
  File "/Users/xxx/.local/pipx/venvs/eth-brownie/lib/python3.10/site-packages/brownie/project/main.py", line 44, in <module>
    from brownie.network import web3
  File "/Users/xxx/.local/pipx/venvs/eth-brownie/lib/python3.10/site-packages/brownie/network/__init__.py", line 4, in <module>
    from .account import Accounts
  File "/Users/xxx/.local/pipx/venvs/eth-brownie/lib/python3.10/site-packages/brownie/network/account.py", line 12, in <module>
    import eth_account
  File "/Users/xxx/.local/pipx/venvs/eth-brownie/lib/python3.10/site-packages/eth_account/__init__.py", line 1, in <module>
    from eth_account.account import (  # noqa: F401
  File "/Users/xxx/.local/pipx/venvs/eth-brownie/lib/python3.10/site-packages/eth_account/account.py", line 8, in <module>
    from cytoolz import (
  File "/Users/xxx/.local/pipx/venvs/eth-brownie/lib/python3.10/site-packages/cytoolz/__init__.py", line 3, in <module>
    from .functoolz import *
ImportError: dlopen(/Users/xxx/.local/pipx/venvs/eth-brownie/lib/python3.10/site-packages/cytoolz/functoolz.cpython-310-darwin.so, 2): Symbol not found: __PyGen_Send
  Referenced from: /Users/xxx/.local/pipx/venvs/eth-brownie/lib/python3.10/site-packages/cytoolz/functoolz.cpython-310-darwin.so
  Expected in: flat namespace
 in /Users/xxx/.local/pipx/venvs/eth-brownie/lib/python3.10/site-packages/cytoolz/functoolz.cpython-310-darwin.so
```

I tried many different way such as:
```
pip install cytoolz
pip install cython 
```

using 
MacOS 10.15.7
python 3.10.2

Do i need to downgrade Python version from 3.10 to 3.8 or 3.9?
If yes, Could you recommend on how to downgrade python version in Mac?
I searched on the internet, but i don't know which way is the best to downgrade my python version. Thanks

**Comment by cromewar** [comment]
Hello there this is the cytools errors, please go to Chronological-issues-from-video, there is a guide to solve this.
Cheers

**Reply by Code1688** [reply]
Hi, Thanks for the reply. I already followed the following link. 
https://medium.com/@cromewar/the-mighty-cytools-error-at-web3-smart-contract-development-with-brownie-63335d50f230

I did install XCode version 12 &  pip3 install Cython. But still same error message

**Reply by cromewar** [reply]
Did you also install cython?

**Reply by Code1688** [reply]
yes by using pip3 install Cython.

```
xxx-MBP:brownie_simple_storage xxx$ pip3 install Cython
Requirement already satisfied: Cython in /Users/xxx/Library/Python/3.10/lib/python/site-packages (0.29.28)
WARNING: You are using pip version 21.2.4; however, version 22.0.3 is available.
You should consider upgrading via the '/Library/Frameworks/Python.framework/Versions/3.10/bin/python3.10 -m pip install --upgrade pip' command.
```

**Reply by cromewar** [reply]
I'm pretty much sure this is due to the python version, 3.10 has a lot of problems with web3 libraries, could please try using a virtualenv with python 3.8 ?

**Reply by Code1688** [reply]
Do you mean downgrade python version to 3.8? May I know how to do that?

**Reply by cromewar** [reply]
You just need to create a virtualenv with that version it's not necessary to uninstall your actual python, as I'm not a Mac user let me do some research which are the commands you would have to use to get it done. Cheers

**Reply by Code1688** [reply]
Thank so much. it means that I will type all the python command in the virtual env of python?

**Reply by cromewar** [reply]
You would like to use something like this:
`virtualenv --python=/usr/bin/python3.8 <path/to/new/virtualenv/>`
Don´t forget to first download and install python 3.8 for mac from here: https://www.python.org/downloads/macos/
